agent/engine/metadataservice/write_metadata.go

```diff
@@ -0,0 +1,88 @@
+package metadataservice
+
+import (
+	"io/ioutil"
+	"os"
+	"regexp"
+
+	"github.com/aws/amazon-ecs-agent/agent/api"
+//	"github.com/cihub/seelog"
+	//docker "github.com/fsouza/go-dockerclient"
+)
+
+const (
+	taskArnRegexExpr = "(:task/)|(arn:aws:ecs:)|:"
```

Comment:
I think we can write a safer parser than this that knows how an ARN is structured.  Probably the easiest way is to `strings.SplitN` with the number of expected `:` characters (which is constant) and grab the last section.